core-bundle/contao/drivers/DC_Folder.php

```diff
@@ -393,7 +390,7 @@ public function showAll()
 									$arrFound[] = $objRoot->path;
 								}
 
-								$arrRoot[] = ($objRoot->type == 'folder') ? $objRoot->path : \dirname($objRoot->path);
+								$arrRoot[] = $objRoot->path;
```

Comment:
Why is it no longer necessary to distinguish between files and folders?

Reply:
Basically, this logic had to be here before because if you restricted `root` to a file (no matter whether manually or via search field) to e.g. `preview.png`, the `DC_Folder` would've displayed nothing because there were no parent trails. We had to use `dirname()` in order to have the parent folder itself visible. This is now obsolete because we always have visible parents 😊